nir/src/main/scala/scala/scalanative/nir/serialization/BinaryDeserializer.scala

```diff
@@ -12,6 +12,8 @@ final class BinaryDeserializer(buffer: ByteBuffer) {
   private val header: Map[Global, Int] = {
     buffer.position(0)
 
+    val hasClInit = getBool
```

Comment:
I believe the meaning of this byte should be generalized from

> there is a static initializer in this file

to

> there is at least one reason to fully load this file even if it's not referenced elsewhere

*At the moment*, this would be equivalent to "there is a clinit", because nothing else could justify having to load the file. But if we add another feature in the IR in the future that would also require the file to be loaded, we would be able to do so without breaking backward binary compatibility. If we stick to "this file has a clinit", then we would have to add *another* byte for the new feature, breaking everything.

Reply:
From the following commit, I believe you already have ta name for this concept: `classesWithEntryPoint`. So this could simply be renamed to `hasEntryPoint`.